src/OAuth1.php

```diff
@@ -335,17 +335,28 @@ public function signRequest($request, $token = null)
      */
     protected function composeSignatureBaseString($method, $url, array $params)
     {
+        $url = isset($url) ? $url : '';
+
         if (strpos($url, '?') !== false) {
             list($url, $queryString) = explode('?', $url, 2);
             parse_str($queryString, $urlParams);
             $params = array_merge($urlParams, $params);
         }
         unset($params['oauth_signature']);
         uksort($params, 'strcmp'); // Parameters are sorted by name, using lexicographical byte value ordering. Ref: Spec: 9.1.1
+
+        $parts0 = strtoupper($method);
+        $parts0 = isset($parts0) ? $parts0 : '';
+
+        $parts1 = isset($url) ? $url : '';
+
+        $parts2 = http_build_query($params, '', '&', PHP_QUERY_RFC3986);
+        $parts2 = isset($parts2) ? $parts2 : '';
```

Comment:
Won't it always be set cause it's defined above?

Reply:
fixed

Reply:
Fixed where?

Reply:
[1](https://github.com/yiisoft/yii2-authclient/pull/383/commits/45cd5305eba0c9f684d529d8f0865c24252bfb72)

[2](https://github.com/yiisoft/yii2-authclient/pull/383/commits/46ccb8617a76595fb651a5510a66eca3c91f724a)